extensions/pkg/controller/healthcheck/reconciler.go

```diff
@@ -141,13 +141,13 @@ func (r *reconciler) performHealthCheck(ctx context.Context, request reconcile.R
 	if err != nil {
 		r.logger.Info("Failed to execute healthChecks. Updating each HealthCheckCondition for the extension resource to ConditionCheckError.", "kind", r.registeredExtension.groupVersionKind.Kind, "health condition types", r.registeredExtension.healthConditionTypes, "name", request.Name, "namespace", request.Namespace, "error", err.Error())
 		for _, healthConditionType := range r.registeredExtension.healthConditionTypes {
-			conditionBuilder, err := gardencorev1beta1helper.NewConditionBuilder(gardencorev1beta1.ConditionType(healthConditionType))
-			if err != nil {
-				return r.resultWithRequeue(), err
+			conditionBuilder, buildErr := gardencorev1beta1helper.NewConditionBuilder(gardencorev1beta1.ConditionType(healthConditionType))
+			if buildErr != nil {
+				return r.resultWithRequeue(), buildErr
 			}
 
-			if err := r.updateExtensionConditionFailedToExecute(ctx, conditionBuilder, healthConditionType, extension, r.registeredExtension.groupVersionKind.Kind, err); err != nil {
-				return r.resultWithRequeue(), err
+			if updateErr := r.updateExtensionConditionFailedToExecute(ctx, conditionBuilder, healthConditionType, extension, r.registeredExtension.groupVersionKind.Kind, err); updateErr != nil {
+				return r.resultWithRequeue(), updateErr
```

Comment:
`RequeueAfter` is not respected when returning non-nil `err`. See https://github.com/gardener/gardener-resource-manager/pull/54#discussion_r427595906.

```suggestion
				return reconcile.Result{}, updateErr
```